starlette/middleware/errors.py

```diff
@@ -1,127 +1,442 @@
+    <div class="snippet">
+        <header>
+            <b><span class="package-dir">{package_dir}</span>{relative_path}</b>
+        </header>
+        <div>
+            {lines}
+        </div>
+        {locals}
+        <footer>
+            <div class="symbol">{package}.{symbol}</div>
+            <div class="package">{package}</div>
+        </footer>
+    </div>
 </div>
-"""  # noqa: E501
+"""
+
+FRAME_LINE_TEMPLATE = """
+<div class="frame {extra_css_class}" id="switch-{id}" data-frame-index="{id}">
+    <div class="dot{dot}"></div>
+    <div>
+        <span class="text-muted">{package}</span>
+        <span title="{file_path}">{file_name}<span>
+        <span class="text-muted">in</span>
+        <span class="symbol">{symbol}</span>
+        <span class="text-muted">at line</span> {line_number}
+    </div>
+</div>
+"""
 
-LINE = """
-<p><span class="frame-line">
-<span class="lineno">{lineno}.</span> {line}</span></p>
+DETAILS_ROW_TEMPLATE = """
+<dl>
+    <dt>{label}</dt>
+    <dd>{value}</dd>
+</dl>
 """
 
-CENTER_LINE = """
-<p class="center-line"><span class="frame-line center-line">
-<span class="lineno">{lineno}.</span> {line}</span></p>
+DETAILS_TEMPLATE = """
+<details {open}>
+    <summary>{label}</summary>
+    {rows}
+</details>
 """
 
 
+def get_relative_filename(path: str) -> str:
+    for sys_path in reversed(sorted(sys.path)):
+        if sys_path in path:
+            path = path.replace(sys_path + "/", "")
+            break
+    return path
+
+
+def format_qual_name(obj: typing.Any) -> str:
+    if inspect.isclass(obj):
+        module_name = obj.__module__
+        type_name = obj.__name__
+    else:
+        module_name = obj.__class__.__module__
+        type_name = obj.__class__.__name__
+
+    if module_name == "builtins":
+        module_name = ""
+    if module_name:
+        return f"{module_name}.{type_name}"
+    return type_name
+
+
+def frame_id(frame: inspect.FrameInfo) -> str:
+    hash = hashlib.md5()
+    hash.update(f"{frame.filename}{frame.lineno}".encode())
+    return hash.hexdigest()
+
+
+def is_vendor(frame: inspect.FrameInfo) -> bool:
+    return sys.exec_prefix in frame.filename
+
+
+def get_package_name(frame: inspect.FrameInfo) -> str:
+    return frame.frame.f_globals.get("__package__", "").split(".")[0]
+
+
+def get_symbol(frame: inspect.FrameInfo) -> str:
+    symbol = ""
+    if "self" in frame.frame.f_locals:
+        symbol = frame.frame.f_locals["self"].__class__.__name__
+
+    if "cls" in frame.frame.f_locals:
+        symbol = frame.frame.f_locals["cls"].__name__
+
+    # if we cannot detect class name then just a method name will be rendered
+    function = frame.function
+    if symbol:
+        function = symbol + "." + function
+    return function
+
+
+def mask_secrets(key: str, value: str) -> str:
+    key = key.lower()
+    if any(
+        [
+            "key" in key,
+            "token" in key,
+            "password" in key,
+            "secret" in key,
+        ]
+    ):
```

Comment:
```suggestion
    if any(k in key for k in ["key", "token", "password", "secret"]):
```
Is this less readable? What do you think?